Add visual differences to new and edit taxon pages

carvil opened the pull request:
It is not clear if we are creating or editing a taxon from a user perspective.
This commit makes sure we have callout divs on the page with color and text
indications, making sure we differentiate between the 2 pages.

Trello: https://trello.com/c/CMul4Ru3/112-improve-how-we-differentiate-between-the-new-taxon-and-edit-taxon-forms-s

The new page looks like this:

<img width="1422" alt="screen shot 2016-08-31 at 10 26 34" src="https://cloud.githubusercontent.com/assets/416701/18123414/993c4474-6f65-11e6-93bb-3bc746fa2eca.png">

The edit page looks like this:

<img width="1418" alt="screen shot 2016-08-31 at 10 26 44" src="https://cloud.githubusercontent.com/assets/416701/18123625/a72d1012-6f66-11e6-86f5-11a14e83e69c.png">

mobaig commented on app/presenters/callout_presenter.rb:
```diff
@@ -0,0 +1,27 @@
+    @title = title
+    @page_type = page_type
+
+    set_callout_setting
```
maybe `configure_callout` or just `callout_settings`.

mobaig commented on app/views/shared/_title.html.erb:
```diff
@@ -0,0 +1,8 @@
+<div class="callout <%= callout.callout_class %>">
+  <div class="callout-title">
+    <%= callout.callout_title %>
```
For clarity, let's either call the variable `callout_presenter` or drop the `callout_` from it's attr_readers.

mobaig commented on spec/features/taxonomy_manager_spec.rb:
```diff
@@ -71,18 +72,27 @@ def when_i_visit_the_taxonomy_page
   end
 
-  def when_i_submit_the_taxon_with_a_title_and_parents
+  def fillin_taxon_form
```
`fill_in_taxon_form`

mobaig commented:
Looks good, some minor naming-related feedback. 👍

mobaig commented:
Come to think of it, I think this PR is pending some further updated following a chat with Mark Hurrell. Will speak to @carvil about this tomorrow when he's back from leave.

carvil commented:
@mobaig thanks for your comments. I've fixed the naming issues and updated the screenshots (they were old and didn't reflect other stuff that went into master). Please let me know what you think.

mobaig commented:
👍
